src/librustc_privacy/lib.rs

```diff
@@ -433,15 +442,6 @@ impl<'b, 'a, 'tcx> TypeVisitor<'tcx> for ReachEverythingInTheInterfaceVisitor<'b
 
         ty.super_visit_with(self)
     }
-
-    fn visit_trait_ref(&mut self, trait_ref: ty::TraitRef<'tcx>) -> bool {
-        if let Some(node_id) = self.ev.tcx.hir.as_local_node_id(trait_ref.def_id) {
-            let item = self.ev.tcx.hir.expect_item(node_id);
```

Comment:
Oh, hmm, so *this* `visit_ty` *did* reach the `visit_trait_ref`. I was hoping the effect is identical, but maybe not.
What if you use `proj.item_def_id` instead of the trait `DefId` for `TyProjection` above?